### PR TITLE
Aric/event to api

### PR DESCRIPTION
**Pull request by AricStewart**

### CODE REVIEW

File: api/src/Controller/Event/PutEvent.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+        $permissions = ['api.put.event'];
+        $this->checkPermissions($permissions);
+
+        $event = $this->getEvent($params['id']);
+        $body = $this->checkRequiredBody($request, []);
+
+        unset($body['cycle']);
+        $cycle_from = $event['cycle'];
+        $cycle_to = $event['cycle'];
+        if (array_key_exists('date_from', $body)) {
```
Comment (mshappe) [inline api/src/Controller/Event/PutEvent.php:77]: These inner blocks feel like they could be factored out into a method.

**Review by mshappe — DISMISSED** [review]
Just some DRY advice, and otherwise, this looks good!

**Comment by AricStewart** [comment]
Good points... Reworked.

File: api/src/Controller/Event/BaseEvent.php [hunk 2]
```diff
@@ -117,5 +118,59 @@ public function __construct(Container $container)
...
+
+        if ($event) {
+            $cycle_from = $event['cycle'];
+            $cycle_to = $event['cycle'];
+        } else {
+            $cycle_from = null;
+            $cycle_to = null;
+        }
+        if (array_key_exists('date_from', $body)) {
+            try {
```
Comment (mshappe) [inline api/src/Controller/Event/BaseEvent.php:134]: Um...not actually seeing any difference to this code...

Comment (AricStewart) [reply api/src/Controller/Event/BaseEvent.php:134]: Oh I missed your point. I thought you meant the blocks between the PUT and POST methods which I did combine.  Not the methods WITHIN the function.  I can take another pass later today if you feel it worth it.

Comment (mshappe) [reply api/src/Controller/Event/BaseEvent.php:134]: Oh, yeah, sorry. Refactor all of that inner logic for handling date to/from to a method, please :-D

Comment (AricStewart) [reply api/src/Controller/Event/BaseEvent.php]: Ok fixed.

**Review by mshappe — APPROVED** [review]
That's the stuff!

**Pull request merged**